h2o-core/src/main/java/hex/ModelMetrics.java

```diff
@@ -60,6 +58,31 @@ private void setModelAndFrameFields(Model model, Frame frame) {
     catch (Throwable t) { }
   }
 
+  public ModelMetrics withModel(Model model) {
+    _modelKey = model == null ? null : model._key;
```

Comment:
Maybe we should not allow overwriting not-null keys - this API could be potentially dangerous.

Perhaps an assert would be appropriate?